node/src/components/block_executor.rs

```diff
@@ -439,16 +442,18 @@ impl<REv: ReactorEventT> Component<REv> for BlockExecutor {
                 effect_builder
                     .get_block_at_height_local(finalized_block.height())
                     .event(move |maybe_block| {
-                        Event::BlockAlreadyExists(maybe_block.is_some(), finalized_block)
+                        Event::BlockAlreadyExists(maybe_block.map(Box::new), finalized_block)
```

Comment:
It feels like we should do that even in the "no deploys" case above?
Also, the event probably shouldn't be called `BlockAlreadyExists`, because it is fired both if it does and doesn't already exist?

I'll make a PR with a suggestion (unless in this case, too, I realize I'm wrong :sweat_smile:).

Reply:
> It feels like we should do that even in the "no deploys" case above?

Definitely. I am surprised with the logic for the `deploys.is_empty` case. It looks like we go into the branch as if there was anything to execute but without any deploys to execute (we unnecessarily checkout the pre-state etc.) where we could simply return early.

Reply:
We need the pre-state even if there is nothing to execute, though, no? We have to create the post-state, which will be identical to the pre-state in this case, but that means we have to know the pre-state.

But you're right - a better place for the shortcut for `get_deploys` could be in the code handling the case when we don't have the block yet in `BlockAlreadyExists`, not here.